picker_wrapper.go

```diff
@@ -120,6 +128,14 @@ func (bp *pickerWrapper) pick(ctx context.Context, failfast bool, opts balancer.
 			bp.mu.Unlock()
 			select {
 			case <-ctx.Done():
+				if pickErr != nil {
```

Comment:
@menghanl PTAL.
Should I check the type of pickErr here? something like:
```
prevErr, ok := pickErr.(balancer.ErrTransientFailure)
```

Reply:
What's the purpose of `pickErr`?

I think it should be OK to always do the following switch and append the connection error.

Reply:
> What's the purpose of `pickErr`?
> 
> I think it should be OK to always do the following switch and append the connection error.

just a flag to judge if `balancer.ErrTransientFailure` occurred.
I will change it.

Reply:
@menghanl PTAL